app/src/main/java/com/github/campus_capture/bootcamp/map/MapScheduler.java

```diff
@@ -129,16 +126,23 @@ public void run() {
      * @param backend the back-end to be used
      * @param upper the fragment above
      */
-    public MapScheduler(View view, FirebaseInterface backend, MapsFragment upper)
+    public MapScheduler(View view, BackendInterface backend, MapsFragment upper)
     {
-        firebaseInterface = backend;
+        backendInterface = backend;
         this.upper = upper;
         scheduledTaskHandler = new Handler();
         zoneText = view.findViewById(R.id.currentZoneText);
         attackButton = view.findViewById(R.id.attackButton);
         defendButton = view.findViewById(R.id.defendButton);
         timerButton = view.findViewById(R.id.timerButton);
-        zoneState = firebaseInterface.getCurrentZoneOwners();
+
+        // TODO not sure how to correctly refactor this using the new future interface
```

Comment:
Technically the scheduler should retry this operation if it fails, I'll keep this in mind the next time I'll be working with the scheduler to add a task for that, for now this should hold up

Reply:
ok perfect